### PR TITLE
Make "User Radio" use favorites instead of local collection

### PR DESCRIPTION
**Pull request by theli-ua**

### CODE REVIEW

**Comment by theli-ua** [comment]
note that user that currently has "upload collection to echonest" has to disable this for catalog to be deleted and then enable it back to recreate (now from favorites)

**Comment by Horrendus** [comment]
Favorites base radio is nice, but I find myself often not favoriting that many songs.
I personally would find it more interesting to base the user radio on:
-) plays
-) skips
-) favorites

**Comment by theli-ua** [comment]
This change is only changing what is used to seed the "user radio"
The way current user radio is implemented its basically useless.

What you're reffering to would require more changes, ideally radio stations would also report back plays/skips/loves/etc to fine-tune stations as echonest supports that.
Also, echonest has a limit on "catalog stations" on non-commercial API keys so unless Tomahawk team is planning to pay echonest putting much more effort into user radio is just not worth it

**Comment by Horrendus** [comment]
I know that only the seed changes and that it currently is seeded by the collection.

I think Tomahawk already saves Plays (not sure about Skips) in the database, so this data can be used to upload (and update from time to time) a "user radio" based on the users listening preferences

File: src/libtomahawk/EchonestCatalogSynchronizer.cpp [hunk 1]
```diff
@@ -54,8 +54,8 @@ EchonestCatalogSynchronizer::EchonestCatalogSynchronizer( QObject *parent )
     qRegisterMetaType<QList<QStringList> >("QList<QStringList>");
 
     connect( TomahawkSettings::instance(), SIGNAL( changed() ), this, SLOT( checkSettingsChanged() ) );
-    connect( SourceList::instance()->getLocal()->dbCollection().data(), SIGNAL( tracksAdded( QList<unsigned int> ) ), this, SLOT( tracksAdded( QList<unsigned int> ) ), Qt::QueuedConnection );
-    connect( SourceList::instance()->getLocal()->dbCollection().data(), SIGNAL( tracksRemoved( QList<unsigned int> ) ), this, SLOT( tracksRemoved( QList<unsigned int> ) ), Qt::QueuedConnection );
+    //connect( SourceList::instance()->getLocal()->dbCollection().data(), SIGNAL( tracksAdded( QList<unsigned int> ) ), this, SLOT( tracksAdded( QList<unsigned int> ) ), Qt::QueuedConnection );
+    //connect( SourceList::instance()->getLocal()->dbCollection().data(), SIGNAL( tracksRemoved( QList<unsigned int> ) ), this, SLOT( tracksRemoved( QList<unsigned int> ) ), Qt::QueuedConnection );
```
Comment (dschmidt) [inline src/libtomahawk/EchonestCatalogSynchronizer.cpp]: Why is this commented out? Either remove it if it's not needed anymore or make it work if it doesn't

File: src/libtomahawk/EchonestCatalogSynchronizer.cpp [hunk 2]
```diff
@@ -176,14 +176,7 @@ EchonestCatalogSynchronizer::songCreateFinished()
...
 
-    QString sql( "SELECT file.id, track.name, artist.name, album.name "
-                 "FROM file, artist, track, file_join "
-                 "LEFT OUTER JOIN album "
-                 "ON file_join.album = album.id "
-                 "WHERE file.id = file_join.file "
-                 "AND file_join.artist = artist.id "
-                 "AND file_join.track = track.id "
-                 "AND file.source IS NULL");
+    QString sql( "SELECT track.id, track.name, artist.name, source FROM social_attributes, track, artist WHERE social_attributes.id = track.id AND artist.id = track.artist AND social_attributes.k = 'Love' AND social_attributes.v = 'true'  AND social_attributes.v = 'true' and social_attributes.source IS NULL GROUP BY track.id");
```
Comment (dschmidt) [inline src/libtomahawk/EchonestCatalogSynchronizer.cpp]: Can you stick to the old style, please?

**Comment by dschmidt** [comment]
I'm +1 on this, once the above mentioned issues are resolved. Thoughts, @muesli?

**Comment by dschmidt** [comment]
@theli-ua with echonest being dead, can we close this too?

**Comment by theli-ua** [comment]
@dschmidt 
Sure, if anything this will still be somewhere in the PR limbo to look it up if needed

**Pull request closed without merging**